pkg/scope/scope.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scope
+
+import (
+	"github.com/go-logr/logr"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/utils/openstack/clientconfig"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Scope struct {
+	ProviderClient     *gophercloud.ProviderClient
+	ProviderClientOpts *clientconfig.ClientOpts
+
+	Logger logr.Logger
+	Client client.Client
```

Comment:
If we're creating the objects in the calling scope and passing a pointer, these should probably remain pointers. Dereferencing these structs in the constructor could get weird.

Reply:
The decision to make the `ProviderClient` members pointers and `Logger` and `Client` values was done based on how these were previously being passed around.

If we remove the `Params` struct, I can just make everything in `Scope` a value and then we pass around the `Scope` as a pointer.